### DISCUSSION TITLE
Sound board (sound control card) communication issue

### DISCUSSION THREAD

**Discussion by hagikent** [harp-tech/protocol#87] [Q&A]
Hi, I set up the driver etc for the sound board as described here: https://bitbucket.org/fchampalimaud/device.soundcard/wiki/Homec

Although I can read what is already written in the board, but seemingly I cannot upload files.
Could someone help me figuring out what could be potentially going wrong here?

1. I can read
![CanRead](https://github.com/harp-tech/harp-tech.github.io/assets/54397345/ccb0729f-f92d-4d25-812f-c5dc2d7344a4)

2. I cannot upload
![IOerror](https://github.com/harp-tech/harp-tech.github.io/assets/54397345/ad9f5c11-1260-45b9-81f6-e27b2d7a458b)

Thanks!
Kenta

**Comment by filcarv** [comment]
Hi Kenta, 

Reading from the device was discontinued in the last firmware revision. This was due to the necessary microprocessor bandwidth to implement the internal sinewave generator. Since reading from the device is something that, at least as far as we know, wasn't being used, we've chosen to deprecate it and release the necessary bandwidth.

When you click **Read from device** in the GUI, the microprocessor ends up entering into an erroneous behavior. A device reset or power cycling solves the erroneous behavior..
